lib/api/bucketGetEncryption.js

```diff
@@ -0,0 +1,84 @@
+const { errors, s3middleware } = require('arsenal');
+
+const bucketShield = require('./apiUtils/bucket/bucketShield');
+const { isBucketAuthorized } =
+    require('./apiUtils/authorization/permissionChecks');
+const metadata = require('../metadata/wrapper');
+const { pushMetric } = require('../utapi/utilities');
+const escapeForXml = s3middleware.escapeForXml;
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+
+const requestType = 'bucketGetEncryption';
+
+/**
+ * Bucket Get Encryption - Get bucket SSE configuration
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+
+function bucketGetEncryption(authInfo, request, log, callback) {
+    const bucketName = request.bucketName;
+    const canonicalID = authInfo.getCanonicalID();
+
+    return metadata.getBucket(bucketName, log, (err, bucket) => {
+        if (err) {
+            log.debug('metadata getbucket failed', { error: err });
+            return callback(err);
+        }
+        if (bucketShield(bucket, requestType)) {
+            return callback(errors.NoSuchBucket);
+        }
+        log.trace('found bucket in metadata');
+
+        const corsHeaders = collectCorsHeaders(request.headers.origin,
+            request.method, bucket);
+
+        if (!isBucketAuthorized(bucket, requestType, canonicalID, authInfo,
+        log)) {
+            log.debug('access denied for account on bucket', {
+                requestType,
+                method: 'bucketGetEncryption',
+            });
+            return callback(errors.AccessDenied, null, corsHeaders);
+        }
+
+        const sseInfo = bucket.getServerSideEncryption();
+
+        // If ssInfo is present but the `mandatory` flag is not set
+        // then this info was not created using bucketPutEncryption
+        // and should not be returned
+        if (sseInfo === null || !sseInfo.mandatory) {
+            log.trace('no server side encryption config found'. {
+                bucket: bucketName,
+                method: 'bucketGetEncryption',
+            });
+            return callback(errors.ServerSideEncryptionConfigurationNotFoundError);
+        }
+
+        const xml = [];
+        xml.push(
+            '<?xml version="1.0" encoding="UTF-8"?>',
+            '<ServerSideEncryptionConfiguration>',
+            '<Rule>',
+            '<ApplyServerSideEncryptionByDefault>',
+            `<KMSMasterKeyID>${escapeForXml(sseInfo.masterKeyId)}</KMSMasterKeyID>`,
```

Comment:
There are two different supported schemes:
- the `aws:kms` mode of encryption: one where the PutBucketEncryption contains a specified master key ID to use for the bucket, in which case it should be returned in the KMSMasterKeyID field
- the `AES256` mode where it uses a managed key (a Scality key) and the user should not know about it as it is meant to be transparent, then we should not return the "KMSMasterKeyID" field even though the bucket has an internal managed key that it uses to encrypt objects.

cf. design doc in citadel and https://docs.aws.amazon.com/AmazonS3/latest/API/API_ServerSideEncryptionByDefault.html